src/Sylius/Bundle/UiBundle/Menu/AbstractMenuBuilder.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\UiBundle\Menu;
+
+use Knp\Menu\FactoryInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+abstract class AbstractMenuBuilder
```

Comment:
All MenuBuilders need factory and event dispatcher, we will have more of them in AdminBundle.